From 77d095cca1383f9f235192517e250cc8aa3f26ce Mon Sep 17 00:00:00 2001
From: Brock Noland <brock@apache.org>
Date: Wed, 28 Jan 2015 19:12:15 +0000
Subject: [PATCH 0003/1164] HIVE-9484 - ThriftCLIService#getDelegationToken
 does case sensitive comparison (Jimmy Xiang via
 Brock)

git-svn-id: https://svn.apache.org/repos/asf/hive/trunk@1655434 13f79535-47bb-0310-9956-ffa450edef68
(cherry picked from commit 59206717608e360bf735f117875e7f55c189a83e)
---
 .../hive/service/cli/thrift/ThriftCLIService.java  |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/service/src/java/org/apache/hive/service/cli/thrift/ThriftCLIService.java b/service/src/java/org/apache/hive/service/cli/thrift/ThriftCLIService.java
index 3345b5f..058e108 100644
--- a/service/src/java/org/apache/hive/service/cli/thrift/ThriftCLIService.java
+++ b/service/src/java/org/apache/hive/service/cli/thrift/ThriftCLIService.java
@@ -347,7 +347,7 @@ SessionHandle getSessionHandle(TOpenSessionReq req, TOpenSessionResp res)
   private String getDelegationToken(String userName)
       throws HiveSQLException, LoginException, IOException {
     if (userName == null || !cliService.getHiveConf().getVar(ConfVars.HIVE_SERVER2_AUTHENTICATION)
-        .equals(HiveAuthFactory.AuthTypes.KERBEROS.toString())) {
+        .equalsIgnoreCase(HiveAuthFactory.AuthTypes.KERBEROS.toString())) {
       return null;
     }
     try {
-- 
1.7.9.5

